src/main/java/uk/gov/pay/api/validation/ValidCountry.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.pay.api.validation;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Target({ FIELD, METHOD, PARAMETER, ANNOTATION_TYPE, TYPE_USE })
+@Retention(RUNTIME)
+@Constraint(validatedBy = ExactLengthValidator.class)
```

Comment:
I've added some comments above, I think it would be better to use `@Size` and `@ReportAsSingleViolation` since its easier to extend in future (such as adding other validations like with did with `@ValidMetadata`). I'm not sure its a good idea to have this annotation validated by a class which is very specific which doesn't have a common name, it was a little confusing at first.

Reply:
So I used `@Size` originally but there was a test about an empty country being valid (https://github.com/alphagov/pay-publicapi/blob/master/src/test/java/uk/gov/pay/api/validation/PaymentRequestValidatorTest.java#L96) and I couldn't use two `@Size`es, one specifying a char length of 2 and one specifying a char length of 0. This is the reason I reused the existing ExactLengthValidator, which should arguably be renamed CountryLengthValidator.

Reply:
There's a sneaky `isBlank(value)` in the ExactLengthValidator that makes an empty country string possible.

Reply:
If it's renamed to CountryValidator, I think it seems fine in that case